osquery/tables/other/augeas.cpp

```
@@ -0,0 +1,50 @@
+#include <osquery/tables.h>
+
+#include <augeas.h>
+#include <boost/algorithm/string/join.hpp>
+
+namespace osquery {
+namespace tables {
+void match(QueryData &results, const augeas *aug, const char *pattern) {
+  char **matches = NULL;
+
+  int len = aug_match(aug, pattern, &matches);
+
+  for (int i = 0; i < len; i++) {
+    Row r;
+    const char *path = matches[i];
```

We can also report an error if `len < 0`. What's the API to do so?